internal/core/src/segcore/SegmentSealedImpl.cpp

```diff
@@ -606,8 +607,38 @@ SegmentSealedImpl::LoadDeletedRecord(const LoadDeletedRecordInfo& info) {
     ParsePksFromIDs(pks, field_meta.get_data_type(), *info.primary_keys);
     auto timestamps = reinterpret_cast<const Timestamp*>(info.timestamps);
 
-    // step 2: fill pks and timestamps
-    deleted_record_.push(pks, timestamps);
+    std::vector<std::tuple<Timestamp, PkType>> ordering(size);
```

Comment:
Can we avoid the unnecessary memory copy here? I suppose by changing `ParsePksFromIDs` it is possible to make the filtering 0-copy.

Reply:
The optimization possible here is to only copy entries matched. The data from protobuf need to copied anyway